codegen/impl/src/main/kotlin/com/yandex/yatagan/codegen/impl/ScopedProviderGenerator.kt

```diff
@@ -79,8 +79,8 @@ internal class ScopedProviderGenerator @Inject constructor(
                                 }
                             }
                         } else {
-                            if (options.enableThreadChecks) {
-                                +"%T.assertThreadAccess()".formatCode(Names.ThreadAssertions)
+                            options.threadCheckerClassName?.let {
```

Comment:
Can we add somewhere some sort of validation of the class?
Previously, we were sure that `assertThreadAccess()` as the class implemented interface.
For now, if user has made any sort of mistake (uses kotlin object without `@JvmStatic`, wrong method name, etc.) he will know about it only on Java compilation step with error like `declaration not found`.

Reply:
Validation is now added